platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesReplayer.java

```diff
@@ -67,19 +71,21 @@ public class PcesReplayer {
      *                                 overwhelmed
      */
     public PcesReplayer(
-            final @NonNull Time time,
+            final @NonNull PlatformContext context,
             final @NonNull StandardOutputWire<PlatformEvent> eventOutputWire,
             final @NonNull Runnable flushIntake,
             final @NonNull Runnable flushTransactionHandling,
             final @NonNull Supplier<ReservedSignedState> latestImmutableState,
             final @NonNull Supplier<Boolean> isSystemHealthy) {
 
-        this.time = Objects.requireNonNull(time);
+        this.time = context.getTime();
```

Comment:
Should we check the context is not null?

Reply:
the `requireNonNull` check simply throws a null pointer exception if the check fails. Since that's the exact behavior of calling a method on `null`, there isn't any explicit check required

Reply:
That is why we should not call `Objects.requireNonNull` without a proper message; it is hard to know if it is a validation or an uncaught error. 
In any case, as you said, both are unexpected scenarios with similar stacktraces.